common/fs/src/cache/tailed_file.rs

```diff
@@ -388,6 +393,29 @@ impl LineMetaMut for LazyLineSerializer {
     }
 }
 
+impl LineBufferMut for LazyLineSerializer {
+    fn get_line_buffer(&mut self) -> Option<&[u8]> {
+        if self.line_buffer.as_ref().is_some() {
+            // Get the value without locking
+            return self.line_buffer.as_deref();
+        }
+
+        match self.reader.try_lock() {
+            Some(file_inner) => {
+                // Cache the value to avoid excessive cloning
+                self.line_buffer = Some(file_inner.buf.clone());
+                self.line_buffer.as_deref()
+            }
+            None => None,
+        }
+    }
+
+    fn set_line_buffer(&mut self, line: Vec<u8>) -> Result<(), LineMetaError> {
+        self.line_buffer = Some(line);
```

Comment:
if line_buffer is Bytes then we can do self.line_buffer = Some(line.into()) and the Vec allocation will be converted straight into a Bytes object without allocating